apps/meteor/client/components/UserAutoCompleteMultiple/UserAutoCompleteMultipleFederated.tsx

```diff
@@ -55,33 +55,55 @@ const UserAutoCompleteMultipleFederated = ({
 		{ keepPreviousData: true },
 	);
 
-	const options = data || [];
+	const options = useMemo(() => data || [], [data]);
 
-	const onAddSelected: ComponentProps<typeof Options>['onSelect'] = ([value]) => {
-		setFilter('');
-		const cachedOption = options.find(([curVal]) => curVal === value)?.[1];
-		if (!cachedOption) {
-			throw new Error('UserAutoCompleteMultiple - onAddSelected - failed to cache option');
-		}
-		setSelectedCache({ ...selectedCache, [value]: cachedOption });
-	};
+	const onAddSelected = useCallback(
+		([value]: OptionType, shouldRemoveUser = false): void => {
+			setFilter('');
+			if (shouldRemoveUser) {
+				return setSelectedCache((users) => {
+					const selectedUsers = { ...users };
+					delete selectedUsers[value];
+					return selectedUsers;
+				});
+			}
+			const cachedOption = options.find(([curVal]) => curVal === value)?.[1];
+			if (!cachedOption) {
+				throw new Error('UserAutoCompleteMultiple - onAddSelected - failed to cache option');
+			}
+			setSelectedCache({ ...selectedCache, [value]: cachedOption });
+		},
+		[selectedCache, setSelectedCache, options],
+	);
+
+	const handleOnChange = useCallback(
+		(usernames: string[]) => {
+			onChange(usernames);
+			const selectedUsernames = Object.entries(selectedCache).map(([, item]) => item.username);
```

Comment:
instead of using selectedCache here, you could use the `value` prop which comes from outside de component. Since it's required in the type definition, should be safe enough.

Also this would make the changes to onAddSelected not needed anymore.